### ISSUE TITLE
Is there a mistake that consider test samples as validating data?

### ISSUE THREAD

**Issue by Imorton-zd** [fchollet/keras#2033]
```
hist = model.fit(train,train_label, batch_size=32, nb_epoch=10, verbose=1, show_accuracy=True, validation_data=(test, test_label))
print (max(hist.history['val_acc']))
```

In my opinion, the codes as above shown, have leveraged the labels of test samples because of early-stop according to the accuracy of validation_data. Am I wrong? `max(hist.history['val_acc'])` cannot be selected as my experimental result?
Should I modify the codes like this:

```
hist = model.fit(train,train_label, batch_size=32, nb_epoch=10, verbose=1, show_accuracy=True, validation_split=0.1)
score, acc = model.evaluate(test, test_label, batch_size=32, show_accuracy=True)
print (acc)
```

However, as far as I am concerned, the finall "acc" is not the best experimental result, because after 10 epochs, the model may be not the best, the best model may be among the 10 epochs.
Please give me some suggestions, many thanks!

**Comment by Imorton-zd** [comment]
@ymcui

**Comment by ersinyar** [comment]
@Imorton-zd Have you come up with an explanation or solution for your problem? Because I am experiencing a similar issue where I split my data into 3 parts, which are training, validation and test. I observe that validation accuracy fluctuates in a small interval; therefore, the last model may not give the best result in terms of accuracy. However, what I expect to see is that val_acc starts from a low value and rise up to higher values. Unfortunately I cannot explain this behaviour. Obviously this is not overfitting also since val_acc does not decrease continuously. Any help would be appreciated.

**Comment by EderSantana** [comment]
yeah, you shouldn't use your test set for validation
the examples are just for illustration.
use your validation data in the validation input and after your method is trained test it with `model.evaluate`
